test/test_telnet.py

```diff
@@ -0,0 +1,101 @@
+import socket
+import threading
+import time
+import pytest
+import subprocess
+import os
+import pty
+
+RFC2217_CMD = bytes([255, 254, 44])  # RFC2217 IAC DONT COM-PORT-OPTION
+
+
+def make_pattern(length):
+    ascii_range = list(range(32, 127))
+    return bytes(ascii_range[i % len(ascii_range)] for i in range(length))
+
+
+@pytest.fixture
+def telnet_recv(cmd):
+    def _recv(buf, timeout=1):
```

Comment:
Define helper functions directly; there is no need to use fixtures for that.

Reply:
Using a fixture for helper functions such as these hides that `telnet_recv` actually uses the `cmd` fixture, making the code unnecessarily hard to read.